### PR TITLE
feature: coordinator support

### PR DESCRIPTION
**Pull request by cyclinder**
---
name: Pull request
about: Tell us about your contribution
labels: ["pr/release/none-required"]
---

---
## Thanks for contributing !

Before submitting a pull request, make sure you read about our Contribution notice here: <https://spidernet-io.github.io/spiderpool/contributing/development/pullrequest/#need-review>

**What this PR does / why we need it**:
feature: coordinator support
**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**make sure your commit is signed off**

### CODE REVIEW

File: cmd/coordinator/cmd/cni_types.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+func validateHwPrefix(prefix string) error {
+	if prefix == "" {
+		return nil
+	}
+	// prefix format like: 00:00、0a:1b
+	matchRegexp, err := regexp.Compile("^" + "(" + "[a-fA-F0-9]{2}[:-][a-fA-F0-9]{2}" + ")" + "$")
+	if err != nil {
+		return err
+	}
+	if !matchRegexp.MatchString(prefix) {
```
Comment (weizhoublue) [inline cmd/coordinator/cmd/cni_types.go:191]: 前缀的 xy:mn 中， y 的最后一 bit 不能是1 ，其代表多播mac， 这种 mac 是无法通信的，应该报错失败

Comment (cyclinder) [reply cmd/coordinator/cmd/cni_types.go:193]: Fixed，`[a-fA-F0-9]{2}[:-][a-fA-F0-9]{2}` ==> `[a-fA-F0-9][a-fA-F,0,2-9][:-][a-fA-F0-9]{2}`

File: charts/spiderpool/templates/daemonset.yaml [hunk 2]
```diff
@@ -137,7 +137,7 @@ spec:
           postStart:
             # Move an eventual old existing binary out of the way, we can't delete it as it might be in use right now
             exec:
-              command: [ "/bin/sh", "-c", "if [ -f '/host/opt/cni/bin/spiderpool' ];then rm -f /host/opt/cni/bin/spiderpool.old ; mv /host/opt/cni/bin/spiderpool /host/opt/cni/bin/spiderpool.old ; fi; cp /usr/bin/spiderpool  /host/opt/cni/bin/spiderpool ; rm -f /host/opt/cni/bin/spiderpool.old &>/dev/null || true " ]
```
Comment (weizhoublue) [inline charts/spiderpool/templates/daemonset.yaml:140]: 可以美化下，以后可能还要 copy
```
BIN_LIST="spiderpool coordinator"
while ITEM in ${BIN_LIST} ; do
	rm -f /host/opt/cni/bin/${ITEM}.old || true
	( [ -f '/host/opt/cni/bin/${ITEM}' ] && mv /host/opt/cni/bin/${ITEM} /host/opt/cni/bin/${ITEM}.old ) || true
	cp /usr/bin/${ITEM} /host/opt/cni/bin/${ITEM}
	rm -f /host/opt/cni/bin/${ITEM}.old &>/dev/null  || true
done
```

Comment (cyclinder) [reply charts/spiderpool/templates/daemonset.yaml:140]: done

**Pull request merged**